src/OpenIddict.EntityFramework/Models/OpenIddictApplication.cs

```diff
@@ -52,6 +59,11 @@ public class OpenIddictApplication<TKey> where TKey : IEquatable<TKey> {
         public virtual string Secret { get; set; }
 
         /// <summary>
+        /// Gets the list of the tokens associated with this application.
+        /// </summary>
+        public virtual IList<TToken> Tokens { get; } = new List<TToken>();
```

Comment:
nope, It's just a minor thing, we ideally don't want to store duplicate tokens. I can live with Collections having a `Count` property that allows negative values, I can live with this as well :trollface: